railties/lib/rails/test_unit/minitest_plugin.rb

```diff
@@ -102,7 +102,34 @@ def self.plugin_rails_init(options)
     reporter << ::Rails::TestUnitReporter.new(options[:io], options)
   end
 
-  mattr_accessor(:run_via) { Hash.new }
+  @@run_via = nil unless defined? @@run_via
+
+  # For backward compatibility, makes it possible to set via `[]=`.
+  def @@run_via.[]=(runner, _)
```

Comment:
Won't this add the `[]=` method to `nil`? I don't see how it can provide backwardscompatibility reliably?